### DISCUSSION TITLE
Is there a mistake in the example code?

### DISCUSSION THREAD

**Discussion by aboode95** [shiro/map2#16] [Q&A] [answered]
In the following code from the example section, how is the `custom_function` function being called without passing the two arguments for `key` and `state`?

```
# use the full power of Python using functions
def custom_function(key, state):
  print("called custom function")

  # custom conditions and complex sequences
  if key == "d":
    return "{ctrl down}a{ctrl up}"
  return True

mapper.map("d", custom_function)
```

**Comment by shiro** [comment, marked as answer]
You pass in the function as a parameter without calling it, `map2` will call it with `key` and `state`.

**Reply by aboode95** [reply]
Ah! thank you! I guess my sleepless brain is affecting my thinking ability xD I should get some sleep~

 Thank you again!
